src/test/ui/consts/const-prop-read-static-in-const.rs

```diff
@@ -0,0 +1,12 @@
+// compile-flags: -Zunleash-the-miri-inside-of-you
+
+#![allow(dead_code)]
+
+const TEST: u8 = MY_STATIC;
+//~^ skipping const checks
+//~| ERROR any use of this value will cause an error
```

Comment:
I actually think this test should *pass*. There is noting wrong with permitting such code, we just haven't written the analysis to do so.

Reply:
I modified the check so this test will pass.

https://github.com/rust-lang/rust/blob/c8eefb66b789437ee156c990ef746d3f324e48d1/src/librustc_mir/interpret/place.rs#L602

Reply:
Uh, no, this is worse. That flag should only affect the type checker, not the interpreter engine.

See https://github.com/rust-lang/rust/pull/64419#discussion_r324698538 for how this could be made to pass: the check should be in `get_static_alloc`, and it should only error out if the static that got access is mutable. Accessing immutable statics from consts is fine as far as const-correctness is concerned.

What I do not understand yet here is, why *do* you want to rule out accesses to statics?